storage/statedb/database_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package statedb
+
+import (
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/storage/database"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var childHash = common.HexToHash("1341655")  // 20190805 in hexadecimal
+var parentHash = common.HexToHash("1343A3F") // 20199999 in hexadecimal
+
+func TestDatabase_Reference(t *testing.T) {
+	memDB := database.NewMemoryDBManager()
+	db := NewDatabaseWithCache(memDB, 128)
+
+	assert.Equal(t, memDB, db.DiskDB())
+	assert.Equal(t, 1, len(db.nodes)) // {} : {}
+
+	db.Reference(childHash, parentHash)
+	assert.Equal(t, 1, len(db.nodes)) // {} : {}
+
+	child := &cachedNode{}
+	parent := &cachedNode{}
+	db.nodes[childHash] = child
+	db.nodes[parentHash] = parent
+
+	db.Reference(childHash, parentHash)
+	assert.Equal(t, 3, len(db.nodes))
+	assert.Equal(t, uint64(1), child.parents)
+	assert.Equal(t, uint64(1), parent.children[childHash])
+
+	db.Reference(childHash, parentHash)
+	assert.Equal(t, 3, len(db.nodes))
+	assert.Equal(t, uint64(1), child.parents)
+	assert.Equal(t, uint64(1), parent.children[childHash])
```

Comment:
I think 49~52 lines are the same with 44~47 lines.

Reply:
It is. This is to check just calling `Reference` does not have any effect.

Reply:
Thanks. It would be better if you add a simple comment about it.

Reply:
Updated!

Reply:
Thank you!